client/js/bar.js

```diff
@@ -59,4 +54,9 @@ function createBarChart(chartValues, chartCategories) {
   }
 }
 
-createBarChart(CHART_VALUES, CHART_CATEGORIES);
+GENRE_ANALYSIS.then((genreAnalysisInfo) => {
```

Comment:
I mentioned this in another review, but consider renaming `GENRE_ANALYSIS` to `GENRE_ANALYSIS_TASK` or `GENRE_ANALYSIS_PROMISE` so that it is extra clear that this object is a `Promise`.